python/hail/linalg/blockmatrix.py

```diff
@@ -553,103 +584,281 @@ def __neg__(self):
         op = getattr(self._jbm, "unary_$minus")
         return BlockMatrix(op())
 
-    @typecheck_method(b=oneof(numeric, block_matrix_type))
+    # scalar => 0, column => 1, row => 2, else => 3
```

Comment:
If you need to do this, you should create and refer to constant definitions.

But ... I think we should try to figure out how to get rid of _shape_type.

Reply:
Also, good first attempt, this shit is very annoying.